### PR TITLE
feat: [DEV-284] Client Library support for User Management - Go

### PR DESCRIPTION
**Pull request by Max-Stytch**

### CODE REVIEW

**Review by jeremy-stytch — APPROVED** [review]
LGTM

**Review by nikhil-stytch — APPROVED** [review]

**Review by grace-stytch — APPROVED** [review]

File: stytch/user.go [hunk 1]
```diff
@@ -100,6 +112,205 @@ type UsersDeleteWebAuthnRegistrationResponse struct {
...
+	CreatedAtLessThan time.Time
+}
+
+func (q UsersSearchQueryCreatedAtLessThanFilter) MarshalJSON() ([]byte, error) {
+	return marshalFilter("created_at_less_than", q.CreatedAtLessThan)
+}
+
+type UsersSearchQueryCreatedAtBetweenFilter struct {
+	CreatedAtGreaterThan time.Time
+	CreatedAtLessThan    time.Time
```
Comment (chris-stytch) [inline stytch/user.go:160]:
```suggestion
	GreaterThan time.Time
	LessThan    time.Time
```

These two shouldn't have the `CreatedAt` prepended.

File: stytch/user.go [hunk 2]
```diff
@@ -100,6 +112,205 @@ type UsersDeleteWebAuthnRegistrationResponse struct {
...
+}
+
+func (q UsersSearchQueryCreatedAtBetweenFilter) MarshalJSON() ([]byte, error) {
+	return marshalFilter("created_at_between", map[string]time.Time{
+		"greater_than": q.CreatedAtGreaterThan,
+		"less_than":    q.CreatedAtLessThan,
```
Comment (chris-stytch) [inline stytch/user.go:166]: Same here

File: stytch/user.go (hunk 1 above)
Comment (Max-Stytch) [reply stytch/user.go]: https://github.com/stytchauth/stytch-go/pull/46/commits/5c03693af1451ef227073d5d89559d64a5e8f0d0

**Pull request merged**